File: OpenEdXMobile/src/androidTest/java/org/edx/mobile/test/feature/matcher/ActionBarMatcher.java
package org.edx.mobile.test.feature.matcher;

import android.view.View;

import org.hamcrest.Matcher;

import static androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA;
import static androidx.test.espresso.matcher.ViewMatchers.withId;

public enum ActionBarMatcher {
    ;

    public static Matcher<View> isInActionBar() {
        return isDescendantOfA(withId(com.google.android.material.R.id.action_bar_container));
    }
}

File: OpenEdXMobile/src/androidTest/java/org/edx/mobile/feature/matcher/ActionBarMatcher.kt
package org.edx.mobile.feature.matcher

import android.view.View
import android.widget.TextView
import androidx.appcompat.widget.Toolbar
import androidx.test.espresso.Espresso
import androidx.test.espresso.ViewInteraction
import androidx.test.espresso.matcher.ViewMatchers
import org.hamcrest.CoreMatchers
import org.hamcrest.Matcher

object ActionBarMatcher {

    @JvmStatic
    fun isInActionBar(): Matcher<View> {
        return ViewMatchers.isDescendantOfA(
            ViewMatchers.withParent(
                ViewMatchers.isAssignableFrom(
                    Toolbar::class.java
                )
            )
        )
    }

    @JvmStatic
    fun isInActionBarInteraction(): ViewInteraction {
        return Espresso.onView(
            CoreMatchers.allOf(
                ViewMatchers.isAssignableFrom(
                    TextView::class.java
                ), ViewMatchers.withParent(
                    ViewMatchers.isAssignableFrom(
                        Toolbar::class.java
                    )
                )
            )
        )
    }
}
